### PR TITLE
certificate-info certificate expiration UI

### LINKED ISSUE #16: Fork of certificate-info for expiry warnings
I have forked certificate-info into https://github.com/kkostrzewa/certificate-info to provide some UI around certificates that are nearing expiry 

Details, along with UI can be found in this world readable google doc: https://docs.google.com/document/d/1vVJrZ9d2s9c2qIDEr85Fh5r6ohE8L_rfNDErsDS2hp4/edit

(I am more than happy to copy that into a github artifact in case you'd rather not open that).

Are you interested in incorporating this back into the main branch? I'm happy to work with you on that

### PR DESCRIPTION
**Pull request by kkostrzewa**
Present a UI around certificate expiration information, as well as badge UI changes when certificates are in a warning mode (45 days from expiry) and error mode (14 days from expiry).

Requires server side modifications to expose certificate expiration information, and client (extension) side modifications to consume that expiration information.

Closes #16

### CODE REVIEW

File: certificate-info/background.js [hunk 1]
```diff
@@ -15,50 +15,55 @@
...
-var colors = {'': '#888', 'gray': '#888', 'red': '#FF1744', 'orange': '#EF6C00'}
+var colors = { '': '#888', 'gray': '#888', 'red': '#FF1744', 'orange': '#EF6C00' }
 
 // In-memory data store
 var cachedValidatonData = {};
 var popupData = {};
 var currentTabId = 0;
 
+// constants around expiration
+const one_day = 1000 * 60 * 60 * 24;
```
Comment (yunzhu-li) [inline certificate-info/background.js:26]: nitpick: can we rename this to be consistent with other variables (camel case)? Sorry I didn't set up a linter.

File: certificate-info/background.js [hunk 2]
```diff
@@ -15,50 +15,55 @@
...
+const expiration_error_threshold_days = 14;      // number of days before certificate management is an ERROR
+const expiration_warning_threshold_days = 45;    // number of days before certificate management is a WARNING
```
[the remark this answers is not in the document]
Comment (kkostrzewa) [reply certificate-info/background.js:28]: I asked some local devops engineers what our policies were around "we should do something" and "we _really_ need to do something" so I came up with these arbitrary windows.

That being said, I'm totally fine with reducing warning to 29 days and leaving error threshold to 14. Sound good?

My original intent was to make these configurable with a UI, but I was admittedly lazy.

File: server/main.go [hunk 3]
```diff
@@ -196,6 +198,7 @@ func validateHost(hostname string) map[string]string {
 	subjectEVOID := ""
 	issuerCommonName := ""
 	issuerOrganization := ""
+	not_after := ""
```
[the remark this answers is not in the document]
Comment (kkostrzewa) [reply server/main.go:201]: no problem. I went back and forth with this, to be consistent with the property name or consistent with the variable name. I will fix.

File: certificate-info/background.js (hunk 1 above)
Comment (kkostrzewa) [reply certificate-info/background.js:26]: also, sounds good. will fix

**Comment by kkostrzewa** [comment]
@yunzhu-li  thanks for the suggestions & reviewing my work. Please forgive me, as this is literally my first time with a true git pull request workflow - I will try to follow reasonable best practices here, but may fumble a bit.

File: certificate-info/background.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kkostrzewa) [reply certificate-info/background.js]: s/b fixed - 29 days

File: server/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kkostrzewa) [reply server/main.go]: s/b fixed: camelCase

File: certificate-info/background.js (hunk 1 above)
Comment (kkostrzewa) [reply certificate-info/background.js]: s/b fixed, camelCase

**Comment by yunzhu-li** [comment]
I'll update the server and Chrome store. Will comment here when done.

**Comment by yunzhu-li** [comment]
Updated. Looks like the new version is available now on the [store](https://chrome.google.com/webstore/detail/certificate-info/jhldepncoippkjgjkmambfglddmjdmaj).

**Comment by kkostrzewa** [comment]
verified that I'm now running 0.2.0 from the store and I see the new functionality. Thanks!

I have some other things I may want to work on here (more prominant alerting and configurable warnings / errors) that I will write issues for, ok?

Thanks again!

**Comment by yunzhu-li** [comment]
Sure. Feel free to open issues and pull requests.

**Pull request merged**